Unitialized Variables in rr_graph_uxsdcxx_serializer.h

<!--- Provide a general summary of the issue in the Title above -->

#### Expected Behaviour
<!--- Tell us what should happen -->
VPR should build without warnings

#### Current Behaviour
<!--- Tell us what happens instead of the expected behaviour -->
With gcc-7:
```
n file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h: In function ‘void uxsd::load_channels(const pugi::xml_node&, T&, Context&, const std::function<void(const char*)>*, ptrdiff_t*) [with T = RrGraphSerializer; Context = void*]’:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1030:36: warning: ‘y_list_info’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         chan_width_->y_list[index] = info;
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2693:21: note: ‘y_list_info’ was declared here
                 int y_list_info;
                     ^~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1026:25: warning: ‘y_list_index’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             report_error(
             ~~~~~~~~~~~~^
                 "index %d on y_list exceeds y_list size %u",
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                 index, chan_width_->y_list.size());
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2692:30: note: ‘y_list_index’ was declared here
                 unsigned int y_list_index;
                              ^~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1003:36: warning: ‘x_list_info’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         chan_width_->x_list[index] = info;
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2685:21: note: ‘x_list_info’ was declared here
                 int x_list_info;
                     ^~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:999:25: warning: ‘x_list_index’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             report_error(
             ~~~~~~~~~~~~^
                 "index %d on x_list exceeds x_list size %u",
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                 index, chan_width_->x_list.size());
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2684:30: note: ‘x_list_index’ was declared here
                 unsigned int x_list_index;
                              ^~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h: In function ‘void uxsd::load_block_types(const pugi::xml_node&, T&, Context&, const std::function<void(const char*)>*, ptrdiff_t*) [with T = RrGraphSerializer; Context = void*]’:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1161:41: warning: ‘pin_ptc’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (block_type_pin_index_to_name(tile, ptc) != value) {
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3070:21: note: ‘pin_ptc’ was declared here
                 int pin_ptc;
                     ^~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1830:9: warning: ‘pin_class_type’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         switch (type) {
         ^~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3149:31: note: ‘pin_class_type’ was declared here
                 enum_pin_type pin_class_type;
                               ^~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1297:9: warning: ‘block_type_width’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (block_info.width != width) {
         ^~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3214:21: note: ‘block_type_width’ was declared here
                 int block_type_width;
                     ^~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1296:60: warning: ‘block_type_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         const auto& block_info = physical_tile_types_.at(id);
                                                            ^
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3213:21: note: ‘block_type_id’ was declared here
                 int block_type_id;
                     ^~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1301:9: warning: ‘block_type_height’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if (block_info.height != height) {
         ^~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3212:21: note: ‘block_type_height’ was declared here
                 int block_type_height;
                     ^~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h: In function ‘void uxsd::load_switch(const pugi::xml_node&, T&, Context&, const std::function<void(const char*)>*, ptrdiff_t*) [with T = RrGraphSerializer; Context = t_rr_switch_inf*]’:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:433:28: warning: ‘sizing_mux_trans_size’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         sw->mux_trans_size = mux_trans_size;
         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2797:23: note: ‘sizing_mux_trans_size’ was declared here
                 float sizing_mux_trans_size;
                       ^~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:432:22: warning: ‘sizing_buf_size’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         sw->buf_size = buf_size;
         ~~~~~~~~~~~~~^~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2796:23: note: ‘sizing_buf_size’ was declared here
                 float sizing_buf_size;
                       ^~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h: In function ‘void uxsd::load_rr_edges(const pugi::xml_node&, T&, Context&, const std::function<void(const char*)>*, ptrdiff_t*) [with T = RrGraphSerializer; Context = void*]’:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:818:33: warning: ‘edge_switch_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             bind.set_edge_target(src_node, sink_node, switch_id);
             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3683:30: note: ‘edge_switch_id’ was declared here
                 unsigned int edge_switch_id;
                              ^~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:811:25: warning: ‘edge_src_node’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             report_error(
             ~~~~~~~~~~~~^
                 "source_node %d is larger than rr_nodes.size() %d",
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                 src_node, rr_nodes_->size());
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3682:30: note: ‘edge_src_node’ was declared here
                 unsigned int edge_src_node;
                              ^~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:818:33: warning: ‘edge_sink_node’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             bind.set_edge_target(src_node, sink_node, switch_id);
             ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3681:30: note: ‘edge_sink_node’ was declared here
                 unsigned int edge_sink_node;
                              ^~~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h: In function ‘void uxsd::load_node(const pugi::xml_node&, T&, Context&, const std::function<void(const char*)>*, ptrdiff_t*) [with T = RrGraphSerializer; Context = int]’:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3518:22: warning: ‘node_segment_segment_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.init_node_segment(context, node_segment_segment_id);
                      ^~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:618:49: warning: ‘node_timing_R’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         node.set_rc_index(find_create_rr_rc_data(R, C));
                           ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3509:23: note: ‘node_timing_R’ was declared here
                 float node_timing_R;
                       ^~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:618:49: warning: ‘node_timing_C’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         node.set_rc_index(find_create_rr_rc_data(R, C));
                           ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3508:23: note: ‘node_timing_C’ was declared here
                 float node_timing_C;
                       ^~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:563:29: warning: ‘node_loc_ylow’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         node.set_coordinates(xlow, ylow, xhigh, yhigh);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3501:21: note: ‘node_loc_ylow’ was declared here
                 int node_loc_ylow;
                     ^~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:563:29: warning: ‘node_loc_yhigh’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         node.set_coordinates(xlow, ylow, xhigh, yhigh);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3500:21: note: ‘node_loc_yhigh’ was declared here
                 int node_loc_yhigh;
                     ^~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:563:29: warning: ‘node_loc_xlow’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         node.set_coordinates(xlow, ylow, xhigh, yhigh);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3499:21: note: ‘node_loc_xlow’ was declared here
                 int node_loc_xlow;
                     ^~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:563:29: warning: ‘node_loc_xhigh’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         node.set_coordinates(xlow, ylow, xhigh, yhigh);
         ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3498:21: note: ‘node_loc_xhigh’ was declared here
                 int node_loc_xhigh;
                     ^~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:564:25: warning: ‘node_loc_ptc’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         node.set_ptc_num(ptc);
         ~~~~~~~~~~~~~~~~^~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3497:21: note: ‘node_loc_ptc’ was declared here
                 int node_loc_ptc;
                     ^~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h: In function ‘void uxsd::load_rr_graph(const pugi::xml_node&, T&, Context&, const std::function<void(const char*)>*, ptrdiff_t*) [with T = RrGraphSerializer; Context = void*]’:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3589:22: warning: ‘node_type’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_rr_nodes_node(context, node_capacity, node_id, node_type);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3587:32: note: ‘node_type’ was declared here
                 enum_node_type node_type;
                                ^~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3589:22: warning: ‘node_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_rr_nodes_node(context, node_capacity, node_id, node_type);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3586:30: note: ‘node_id’ was declared here
                 unsigned int node_id;
                              ^~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3589:22: warning: ‘node_capacity’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_rr_nodes_node(context, node_capacity, node_id, node_type);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3585:30: note: ‘node_capacity’ was declared here
                 unsigned int node_capacity;
                              ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3296:22: warning: ‘grid_loc_y’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_grid_locs_grid_loc(context, grid_loc_block_type_id, grid_loc_height_offset, grid_loc_width_offset, grid_loc_x, grid_loc_y);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3294:21: note: ‘grid_loc_y’ was declared here
                 int grid_loc_y;
                     ^~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3296:22: warning: ‘grid_loc_x’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_grid_locs_grid_loc(context, grid_loc_block_type_id, grid_loc_height_offset, grid_loc_width_offset, grid_loc_x, grid_loc_y);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3293:21: note: ‘grid_loc_x’ was declared here
                 int grid_loc_x;
                     ^~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3296:22: warning: ‘grid_loc_width_offset’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_grid_locs_grid_loc(context, grid_loc_block_type_id, grid_loc_height_offset, grid_loc_width_offset, grid_loc_x, grid_loc_y);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3292:21: note: ‘grid_loc_width_offset’ was declared here
                 int grid_loc_width_offset;
                     ^~~~~~~~~~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3296:22: warning: ‘grid_loc_height_offset’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_grid_locs_grid_loc(context, grid_loc_block_type_id, grid_loc_height_offset, grid_loc_width_offset, grid_loc_x, grid_loc_y);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3291:21: note: ‘grid_loc_height_offset’ was declared here
                 int grid_loc_height_offset;
                     ^~~~~~~~~~~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3296:22: warning: ‘grid_loc_block_type_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
                 auto child_context = out.add_grid_locs_grid_loc(context, grid_loc_block_type_id, grid_loc_height_offset, grid_loc_width_offset, grid_loc_x, grid_loc_y);
                      ^~~~~~~~~~~~~
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:3290:21: note: ‘grid_loc_block_type_id’ was declared here
                 int grid_loc_block_type_id;
                     ^~~~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:1131:35: warning: ‘segment_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         return &segment_inf_.at(id);
                                   ^
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2996:21: note: ‘segment_id’ was declared here
                 int segment_id;
                     ^~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:18:0:
/project/trees/vtr/vpr/src/route/rr_graph_uxsdcxx_serializer.h:464:28: warning: ‘switch_id’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         make_room_in_vector(rr_switch_inf_, id);
         ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
In file included from /project/trees/vtr/vpr/src/route/rr_graph_reader.cpp:19:0:
/project/trees/vtr/vpr/src/route/gen/rr_graph_uxsdcxx.h:2861:21: note: ‘switch_id’ was declared here
                 int switch_id;
                     ^~~~~~~~~
```


#### Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug -->
Since this is in the generated code, it seems like the uxsdcxx generator will need to be updated to ensure these variables are always initialized.

#### Steps to Reproduce
<!--- Provide an unambiguous set of steps to reproduce this bug. -->
<!--- Include the exact command line arguments and source files -->
<!-- (e.g. architecture file, netlist) to reproduce, if relevant. -->
1. Checkout  revision: 868126ec9545344db9b36a02534d7ca7814b1a97 (current master)
2. make vpr


#### Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->
VTR should build warning clean, so that any newly introduced warnings are quickly identified and fixed. Keeping even otherwise innocuous warnings around makes it much more difficult for developers to do this.

#### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in. -->
* VTR revision used: 868126ec9545344db9b36a02534d7ca7814b1a97
* Operating System and version: Ubuntu 18.04
* Compiler version: gcc-7

--- a/libs/libarchfpga/src/physical_types.h
+++ b/libs/libarchfpga/src/physical_types.h
@@ -93,7 +93,7 @@ class t_metadata_value {
   public:
     explicit t_metadata_value(vtr::interned_string v)
         : value_(v) {}
-    explicit t_metadata_value(const t_metadata_value& o)
+    explicit t_metadata_value(const t_metadata_value& o) noexcept
         : value_(o.value_) {}
 
     // Return string value.

--- a/vpr/src/route/gen/rr_graph_uxsdcxx.h
+++ b/vpr/src/route/gen/rr_graph_uxsdcxx.h
@@ -35,6 +35,11 @@ namespace uxsd {
  */
 inline void get_line_number(const char* filename, std::ptrdiff_t offset, int* line, int* col);
 
+[[noreturn]] inline void noreturn_report(const std::function<void(const char*)>* report_error, const char* msg) {
+    (*report_error)(msg);
+    throw std::runtime_error("Unreachable!");
+}
+
 /* Declarations for internal load functions for the complex types. */
 template<class T, typename Context>
 inline void load_channel(const pugi::xml_node& root, T& out, Context& context, const std::function<void(const char*)>* report_error, ptrdiff_t* offset_debug);
@@ -423,8 +428,7 @@ inline atok_t_channel lex_attr_t_channel(const char* in, const std::function<voi
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <channel>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <channel>.").c_str());
 }
 
 inline atok_t_x_list lex_attr_t_x_list(const char* in, const std::function<void(const char*)>* report_error) {
@@ -457,8 +461,7 @@ inline atok_t_x_list lex_attr_t_x_list(const char* in, const std::function<void(
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <x_list>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <x_list>.").c_str());
 }
 
 inline atok_t_y_list lex_attr_t_y_list(const char* in, const std::function<void(const char*)>* report_error) {
@@ -491,8 +494,7 @@ inline atok_t_y_list lex_attr_t_y_list(const char* in, const std::function<void(
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <y_list>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <y_list>.").c_str());
 }
 
 inline gtok_t_channels lex_node_t_channels(const char* in, const std::function<void(const char*)>* report_error) {
@@ -564,8 +566,7 @@ inline gtok_t_channels lex_node_t_channels(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <channels>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <channels>.").c_str());
 }
 
 inline atok_t_timing lex_attr_t_timing(const char* in, const std::function<void(const char*)>* report_error) {
@@ -631,8 +632,7 @@ inline atok_t_timing lex_attr_t_timing(const char* in, const std::function<void(
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <timing>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <timing>.").c_str());
 }
 
 inline atok_t_sizing lex_attr_t_sizing(const char* in, const std::function<void(const char*)>* report_error) {
@@ -677,8 +677,7 @@ inline atok_t_sizing lex_attr_t_sizing(const char* in, const std::function<void(
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <sizing>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <sizing>.").c_str());
 }
 
 inline gtok_t_switch lex_node_t_switch(const char* in, const std::function<void(const char*)>* report_error) {
@@ -723,8 +722,7 @@ inline gtok_t_switch lex_node_t_switch(const char* in, const std::function<void(
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <switch>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <switch>.").c_str());
 }
 inline atok_t_switch lex_attr_t_switch(const char* in, const std::function<void(const char*)>* report_error) {
     unsigned int len = strlen(in);
@@ -759,8 +757,7 @@ inline atok_t_switch lex_attr_t_switch(const char* in, const std::function<void(
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <switch>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <switch>.").c_str());
 }
 
 inline gtok_t_switches lex_node_t_switches(const char* in, const std::function<void(const char*)>* report_error) {
@@ -790,8 +787,7 @@ inline gtok_t_switches lex_node_t_switches(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <switches>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <switches>.").c_str());
 }
 
 inline atok_t_segment_timing lex_attr_t_segment_timing(const char* in, const std::function<void(const char*)>* report_error) {
@@ -848,8 +844,7 @@ inline atok_t_segment_timing lex_attr_t_segment_timing(const char* in, const std
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <segment_timing>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <segment_timing>.").c_str());
 }
 
 inline gtok_t_segment lex_node_t_segment(const char* in, const std::function<void(const char*)>* report_error) {
@@ -879,8 +874,7 @@ inline gtok_t_segment lex_node_t_segment(const char* in, const std::function<voi
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <segment>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <segment>.").c_str());
 }
 inline atok_t_segment lex_attr_t_segment(const char* in, const std::function<void(const char*)>* report_error) {
     unsigned int len = strlen(in);
@@ -912,8 +906,7 @@ inline atok_t_segment lex_attr_t_segment(const char* in, const std::function<voi
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <segment>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <segment>.").c_str());
 }
 
 inline gtok_t_segments lex_node_t_segments(const char* in, const std::function<void(const char*)>* report_error) {
@@ -949,8 +942,7 @@ inline gtok_t_segments lex_node_t_segments(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <segments>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <segments>.").c_str());
 }
 
 inline atok_t_pin lex_attr_t_pin(const char* in, const std::function<void(const char*)>* report_error) {
@@ -980,8 +972,7 @@ inline atok_t_pin lex_attr_t_pin(const char* in, const std::function<void(const 
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <pin>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <pin>.").c_str());
 }
 
 inline gtok_t_pin_class lex_node_t_pin_class(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1011,8 +1002,7 @@ inline gtok_t_pin_class lex_node_t_pin_class(const char* in, const std::function
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <pin_class>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <pin_class>.").c_str());
 }
 inline atok_t_pin_class lex_attr_t_pin_class(const char* in, const std::function<void(const char*)>* report_error) {
     unsigned int len = strlen(in);
@@ -1029,8 +1019,7 @@ inline atok_t_pin_class lex_attr_t_pin_class(const char* in, const std::function
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <pin_class>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <pin_class>.").c_str());
 }
 
 inline gtok_t_block_type lex_node_t_block_type(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1054,8 +1043,7 @@ inline gtok_t_block_type lex_node_t_block_type(const char* in, const std::functi
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <block_type>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <block_type>.").c_str());
 }
 inline atok_t_block_type lex_attr_t_block_type(const char* in, const std::function<void(const char*)>* report_error) {
     unsigned int len = strlen(in);
@@ -1123,8 +1111,7 @@ inline atok_t_block_type lex_attr_t_block_type(const char* in, const std::functi
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <block_type>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <block_type>.").c_str());
 }
 
 inline gtok_t_block_types lex_node_t_block_types(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1154,8 +1141,7 @@ inline gtok_t_block_types lex_node_t_block_types(const char* in, const std::func
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <block_types>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <block_types>.").c_str());
 }
 
 inline atok_t_grid_loc lex_attr_t_grid_loc(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1227,8 +1213,7 @@ inline atok_t_grid_loc lex_attr_t_grid_loc(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <grid_loc>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <grid_loc>.").c_str());
 }
 
 inline gtok_t_grid_locs lex_node_t_grid_locs(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1246,8 +1231,7 @@ inline gtok_t_grid_locs lex_node_t_grid_locs(const char* in, const std::function
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <grid_locs>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <grid_locs>.").c_str());
 }
 
 inline atok_t_node_loc lex_attr_t_node_loc(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1316,8 +1300,7 @@ inline atok_t_node_loc lex_attr_t_node_loc(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <node_loc>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <node_loc>.").c_str());
 }
 
 inline atok_t_node_timing lex_attr_t_node_timing(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1338,8 +1321,7 @@ inline atok_t_node_timing lex_attr_t_node_timing(const char* in, const std::func
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <node_timing>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <node_timing>.").c_str());
 }
 
 inline atok_t_node_segment lex_attr_t_node_segment(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1369,8 +1351,7 @@ inline atok_t_node_segment lex_attr_t_node_segment(const char* in, const std::fu
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <node_segment>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <node_segment>.").c_str());
 }
 
 inline atok_t_meta lex_attr_t_meta(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1388,8 +1369,7 @@ inline atok_t_meta lex_attr_t_meta(const char* in, const std::function<void(cons
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <meta>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <meta>.").c_str());
 }
 
 inline gtok_t_metadata lex_node_t_metadata(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1407,8 +1387,7 @@ inline gtok_t_metadata lex_node_t_metadata(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <metadata>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <metadata>.").c_str());
 }
 
 inline gtok_t_node lex_node_t_node(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1495,8 +1474,7 @@ inline gtok_t_node lex_node_t_node(const char* in, const std::function<void(cons
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <node>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <node>.").c_str());
 }
 inline atok_t_node lex_attr_t_node(const char* in, const std::function<void(const char*)>* report_error) {
     unsigned int len = strlen(in);
@@ -1552,8 +1530,7 @@ inline atok_t_node lex_attr_t_node(const char* in, const std::function<void(cons
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <node>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <node>.").c_str());
 }
 
 inline gtok_t_rr_nodes lex_node_t_rr_nodes(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1571,8 +1548,7 @@ inline gtok_t_rr_nodes lex_node_t_rr_nodes(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <rr_nodes>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <rr_nodes>.").c_str());
 }
 
 inline gtok_t_edge lex_node_t_edge(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1590,8 +1566,7 @@ inline gtok_t_edge lex_node_t_edge(const char* in, const std::function<void(cons
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <edge>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <edge>.").c_str());
 }
 inline atok_t_edge lex_attr_t_edge(const char* in, const std::function<void(const char*)>* report_error) {
     unsigned int len = strlen(in);
@@ -1632,8 +1607,7 @@ inline atok_t_edge lex_attr_t_edge(const char* in, const std::function<void(cons
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <edge>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <edge>.").c_str());
 }
 
 inline gtok_t_rr_edges lex_node_t_rr_edges(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1651,8 +1625,7 @@ inline gtok_t_rr_edges lex_node_t_rr_edges(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <rr_edges>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <rr_edges>.").c_str());
 }
 
 inline gtok_t_rr_graph lex_node_t_rr_graph(const char* in, const std::function<void(const char*)>* report_error) {
@@ -1718,8 +1691,7 @@ inline gtok_t_rr_graph lex_node_t_rr_graph(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized child " + std::string(in) + " of <rr_graph>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized child " + std::string(in) + " of <rr_graph>.").c_str());
 }
 inline atok_t_rr_graph lex_attr_t_rr_graph(const char* in, const std::function<void(const char*)>* report_error) {
     unsigned int len = strlen(in);
@@ -1766,26 +1738,25 @@ inline atok_t_rr_graph lex_attr_t_rr_graph(const char* in, const std::function<v
         default:
             break;
     }
-    (*report_error)(("Found unrecognized attribute " + std::string(in) + " of <rr_graph>.").c_str());
-    throw std::runtime_error("Unreachable code!");
+    noreturn_report(report_error, ("Found unrecognized attribute " + std::string(in) + " of <rr_graph>.").c_str());
 }
 
 /**
  * Internal error function for xs:choice and xs:sequence validators.
  */
-inline void dfa_error(const char* wrong, const int* states, const char* const* lookup, int len, const std::function<void(const char*)>* report_error);
+[[noreturn]] inline void dfa_error(const char* wrong, const int* states, const char* const* lookup, int len, const std::function<void(const char*)>* report_error);
 
 /**
  * Internal error function for xs:all validators.
  */
 template<std::size_t N>
-inline void all_error(std::bitset<N> gstate, const char* const* lookup, const std::function<void(const char*)>* report_error);
+[[noreturn]] inline void all_error(std::bitset<N> gstate, const char* const* lookup, const std::function<void(const char*)>* report_error);
 
 /**
  * Internal error function for attribute validators.
  */
 template<std::size_t N>
-inline void attr_error(std::bitset<N> astate, const char* const* lookup, const std::function<void(const char*)>* report_error);
+[[noreturn]] inline void attr_error(std::bitset<N> astate, const char* const* lookup, const std::function<void(const char*)>* report_error);
 
 /* Lookup tables for enums. */
 constexpr const char* lookup_switch_type[] = {"UXSD_INVALID", "mux", "tristate", "pass_gate", "short", "buffer"};
@@ -1883,7 +1854,7 @@ inline enum_switch_type lex_enum_switch_type(const char* in, bool throw_on_inval
             break;
     }
     if (throw_on_invalid)
-        (*report_error)(("Found unrecognized enum value " + std::string(in) + " of enum_switch_type.").c_str());
+        noreturn_report(report_error, ("Found unrecognized enum value " + std::string(in) + " of enum_switch_type.").c_str());
     return enum_switch_type::UXSD_INVALID;
 }
 
@@ -1939,7 +1910,7 @@ inline enum_pin_type lex_enum_pin_type(const char* in, bool throw_on_invalid, co
             break;
     }
     if (throw_on_invalid)
-        (*report_error)(("Found unrecognized enum value " + std::string(in) + " of enum_pin_type.").c_str());
+        noreturn_report(report_error, ("Found unrecognized enum value " + std::string(in) + " of enum_pin_type.").c_str());
     return enum_pin_type::UXSD_INVALID;
 }
 
@@ -2004,7 +1975,7 @@ inline enum_node_type lex_enum_node_type(const char* in, bool throw_on_invalid, 
             break;
     }
     if (throw_on_invalid)
-        (*report_error)(("Found unrecognized enum value " + std::string(in) + " of enum_node_type.").c_str());
+        noreturn_report(report_error, ("Found unrecognized enum value " + std::string(in) + " of enum_node_type.").c_str());
     return enum_node_type::UXSD_INVALID;
 }
 
@@ -2084,7 +2055,7 @@ inline enum_node_direction lex_enum_node_direction(const char* in, bool throw_on
             break;
     }
     if (throw_on_invalid)
-        (*report_error)(("Found unrecognized enum value " + std::string(in) + " of enum_node_direction.").c_str());
+        noreturn_report(report_error, ("Found unrecognized enum value " + std::string(in) + " of enum_node_direction.").c_str());
     return enum_node_direction::UXSD_INVALID;
 }
 
@@ -2161,7 +2132,7 @@ inline enum_loc_side lex_enum_loc_side(const char* in, bool throw_on_invalid, co
             break;
     }
     if (throw_on_invalid)
-        (*report_error)(("Found unrecognized enum value " + std::string(in) + " of enum_loc_side.").c_str());
+        noreturn_report(report_error, ("Found unrecognized enum value " + std::string(in) + " of enum_loc_side.").c_str());
     return enum_loc_side::UXSD_INVALID;
 }
 
@@ -2170,7 +2141,7 @@ inline int load_int(const char* in, const std::function<void(const char*)>* repo
     int out;
     out = std::strtol(in, NULL, 10);
     if (errno != 0)
-        (*report_error)(("Invalid value `" + std::string(in) + "` when loading into a int.").c_str());
+        noreturn_report(report_error, ("Invalid value `" + std::string(in) + "` when loading into a int.").c_str());
     return out;
 }
 
@@ -2178,7 +2149,7 @@ inline unsigned int load_unsigned_int(const char* in, const std::function<void(c
     unsigned int out;
     out = std::strtoul(in, NULL, 10);
     if (errno != 0)
-        (*report_error)(("Invalid value `" + std::string(in) + "` when loading into a unsigned int.").c_str());
+        noreturn_report(report_error, ("Invalid value `" + std::string(in) + "` when loading into a unsigned int.").c_str());
     return out;
 }
 
@@ -2186,7 +2157,7 @@ inline float load_float(const char* in, const std::function<void(const char*)>* 
     float out;
     out = std::strtof(in, NULL);
     if (errno != 0)
-        (*report_error)(("Invalid value `" + std::string(in) + "` when loading into a float.").c_str());
+        noreturn_report(report_error, ("Invalid value `" + std::string(in) + "` when loading into a float.").c_str());
     return out;
 }
 inline void load_channel_required_attributes(const pugi::xml_node& root, int* chan_width_max, int* x_max, int* x_min, int* y_max, int* y_min, const std::function<void(const char*)>* report_error) {
@@ -2196,7 +2167,7 @@ inline void load_channel_required_attributes(const pugi::xml_node& root, int* ch
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <channel>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <channel>.").c_str());
         switch (in) {
             case atok_t_channel::CHAN_WIDTH_MAX:
                 *chan_width_max = load_int(attr.value(), report_error);
@@ -2228,7 +2199,7 @@ inline void load_x_list_required_attributes(const pugi::xml_node& root, unsigned
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <x_list>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <x_list>.").c_str());
         switch (in) {
             case atok_t_x_list::INDEX:
                 *index = load_unsigned_int(attr.value(), report_error);
@@ -2251,7 +2222,7 @@ inline void load_y_list_required_attributes(const pugi::xml_node& root, unsigned
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <y_list>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <y_list>.").c_str());
         switch (in) {
             case atok_t_y_list::INDEX:
                 *index = load_unsigned_int(attr.value(), report_error);
@@ -2274,7 +2245,7 @@ inline void load_sizing_required_attributes(const pugi::xml_node& root, float* b
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <sizing>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <sizing>.").c_str());
         switch (in) {
             case atok_t_sizing::BUF_SIZE:
                 *buf_size = load_float(attr.value(), report_error);
@@ -2297,7 +2268,7 @@ inline void load_switch_required_attributes(const pugi::xml_node& root, int* id,
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <switch>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <switch>.").c_str());
         switch (in) {
             case atok_t_switch::ID:
                 *id = load_int(attr.value(), report_error);
@@ -2323,7 +2294,7 @@ inline void load_segment_required_attributes(const pugi::xml_node& root, int* id
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <segment>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <segment>.").c_str());
         switch (in) {
             case atok_t_segment::ID:
                 *id = load_int(attr.value(), report_error);
@@ -2346,7 +2317,7 @@ inline void load_pin_required_attributes(const pugi::xml_node& root, int* ptc, c
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <pin>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <pin>.").c_str());
         switch (in) {
             case atok_t_pin::PTC:
                 *ptc = load_int(attr.value(), report_error);
@@ -2366,7 +2337,7 @@ inline void load_pin_class_required_attributes(const pugi::xml_node& root, enum_
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <pin_class>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <pin_class>.").c_str());
         switch (in) {
             case atok_t_pin_class::TYPE:
                 *type = lex_enum_pin_type(attr.value(), true, report_error);
@@ -2386,7 +2357,7 @@ inline void load_block_type_required_attributes(const pugi::xml_node& root, int*
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <block_type>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <block_type>.").c_str());
         switch (in) {
             case atok_t_block_type::HEIGHT:
                 *height = load_int(attr.value(), report_error);
@@ -2415,7 +2386,7 @@ inline void load_grid_loc_required_attributes(const pugi::xml_node& root, int* b
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <grid_loc>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <grid_loc>.").c_str());
         switch (in) {
             case atok_t_grid_loc::BLOCK_TYPE_ID:
                 *block_type_id = load_int(attr.value(), report_error);
@@ -2447,7 +2418,7 @@ inline void load_node_loc_required_attributes(const pugi::xml_node& root, int* p
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <node_loc>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <node_loc>.").c_str());
         switch (in) {
             case atok_t_node_loc::PTC:
                 *ptc = load_int(attr.value(), report_error);
@@ -2482,7 +2453,7 @@ inline void load_node_timing_required_attributes(const pugi::xml_node& root, flo
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <node_timing>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <node_timing>.").c_str());
         switch (in) {
             case atok_t_node_timing::C:
                 *C = load_float(attr.value(), report_error);
@@ -2505,7 +2476,7 @@ inline void load_node_segment_required_attributes(const pugi::xml_node& root, in
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <node_segment>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <node_segment>.").c_str());
         switch (in) {
             case atok_t_node_segment::SEGMENT_ID:
                 *segment_id = load_int(attr.value(), report_error);
@@ -2525,7 +2496,7 @@ inline void load_node_required_attributes(const pugi::xml_node& root, unsigned i
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <node>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <node>.").c_str());
         switch (in) {
             case atok_t_node::CAPACITY:
                 *capacity = load_unsigned_int(attr.value(), report_error);
@@ -2554,7 +2525,7 @@ inline void load_edge_required_attributes(const pugi::xml_node& root, unsigned i
         if (astate[(int)in] == 0)
             astate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate attribute " + std::string(attr.name()) + " in <edge>.").c_str());
+            noreturn_report(report_error, ("Duplicate attribute " + std::string(attr.name()) + " in <edge>.").c_str());
         switch (in) {
             case atok_t_edge::SINK_NODE:
                 *sink_node = load_unsigned_int(attr.value(), report_error);
@@ -2582,7 +2553,7 @@ inline void load_channel(const pugi::xml_node& root, T& out, Context& context, c
     *offset_debug = root.offset_debug();
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <channel>.");
+        noreturn_report(report_error, "Unexpected child element in <channel>.");
 }
 
 template<class T, typename Context>
@@ -2595,7 +2566,7 @@ inline void load_x_list(const pugi::xml_node& root, T& out, Context& context, co
     *offset_debug = root.offset_debug();
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <x_list>.");
+        noreturn_report(report_error, "Unexpected child element in <x_list>.");
 }
 
 template<class T, typename Context>
@@ -2608,7 +2579,7 @@ inline void load_y_list(const pugi::xml_node& root, T& out, Context& context, co
     *offset_debug = root.offset_debug();
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <y_list>.");
+        noreturn_report(report_error, "Unexpected child element in <y_list>.");
 }
 
 constexpr int NUM_T_CHANNELS_STATES = 4;
@@ -2629,7 +2600,7 @@ inline void load_channels(const pugi::xml_node& root, T& out, Context& context, 
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <channels>.");
+        noreturn_report(report_error, "Unexpected attribute in <channels>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t x_list_count = 0;
@@ -2671,10 +2642,15 @@ inline void load_channels(const pugi::xml_node& root, T& out, Context& context, 
         switch (in) {
             case gtok_t_channels::CHANNEL: {
                 int channel_chan_width_max;
+                memset(&channel_chan_width_max, 0, sizeof(channel_chan_width_max));
                 int channel_x_max;
+                memset(&channel_x_max, 0, sizeof(channel_x_max));
                 int channel_x_min;
+                memset(&channel_x_min, 0, sizeof(channel_x_min));
                 int channel_y_max;
+                memset(&channel_y_max, 0, sizeof(channel_y_max));
                 int channel_y_min;
+                memset(&channel_y_min, 0, sizeof(channel_y_min));
                 load_channel_required_attributes(node, &channel_chan_width_max, &channel_x_max, &channel_x_min, &channel_y_max, &channel_y_min, report_error);
                 auto child_context = out.init_channels_channel(context, channel_chan_width_max, channel_x_max, channel_x_min, channel_y_max, channel_y_min);
                 load_channel(node, out, child_context, report_error, offset_debug);
@@ -2682,7 +2658,9 @@ inline void load_channels(const pugi::xml_node& root, T& out, Context& context, 
             } break;
             case gtok_t_channels::X_LIST: {
                 unsigned int x_list_index;
+                memset(&x_list_index, 0, sizeof(x_list_index));
                 int x_list_info;
+                memset(&x_list_info, 0, sizeof(x_list_info));
                 load_x_list_required_attributes(node, &x_list_index, &x_list_info, report_error);
                 auto child_context = out.add_channels_x_list(context, x_list_index, x_list_info);
                 load_x_list(node, out, child_context, report_error, offset_debug);
@@ -2690,7 +2668,9 @@ inline void load_channels(const pugi::xml_node& root, T& out, Context& context, 
             } break;
             case gtok_t_channels::Y_LIST: {
                 unsigned int y_list_index;
+                memset(&y_list_index, 0, sizeof(y_list_index));
                 int y_list_info;
+                memset(&y_list_info, 0, sizeof(y_list_info));
                 load_y_list_required_attributes(node, &y_list_index, &y_list_info, report_error);
                 auto child_context = out.add_channels_y_list(context, y_list_index, y_list_info);
                 load_y_list(node, out, child_context, report_error, offset_debug);
@@ -2736,7 +2716,7 @@ inline void load_timing(const pugi::xml_node& root, T& out, Context& context, co
     }
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <timing>.");
+        noreturn_report(report_error, "Unexpected child element in <timing>.");
 }
 
 template<class T, typename Context>
@@ -2749,7 +2729,7 @@ inline void load_sizing(const pugi::xml_node& root, T& out, Context& context, co
     *offset_debug = root.offset_debug();
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <sizing>.");
+        noreturn_report(report_error, "Unexpected child element in <sizing>.");
 }
 
 template<class T, typename Context>
@@ -2785,7 +2765,7 @@ inline void load_switch(const pugi::xml_node& root, T& out, Context& context, co
         if (gstate[(int)in] == 0)
             gstate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate element " + std::string(node.name()) + " in <switch>.").c_str());
+            noreturn_report(report_error, ("Duplicate element " + std::string(node.name()) + " in <switch>.").c_str());
         switch (in) {
             case gtok_t_switch::TIMING: {
                 auto child_context = out.init_switch_timing(context);
@@ -2794,7 +2774,9 @@ inline void load_switch(const pugi::xml_node& root, T& out, Context& context, co
             } break;
             case gtok_t_switch::SIZING: {
                 float sizing_buf_size;
+                memset(&sizing_buf_size, 0, sizeof(sizing_buf_size));
                 float sizing_mux_trans_size;
+                memset(&sizing_mux_trans_size, 0, sizeof(sizing_mux_trans_size));
                 load_sizing_required_attributes(node, &sizing_buf_size, &sizing_mux_trans_size, report_error);
                 auto child_context = out.init_switch_sizing(context, sizing_buf_size, sizing_mux_trans_size);
                 load_sizing(node, out, child_context, report_error, offset_debug);
@@ -2824,7 +2806,7 @@ inline void load_switches(const pugi::xml_node& root, T& out, Context& context, 
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <switches>.");
+        noreturn_report(report_error, "Unexpected attribute in <switches>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t switch_count = 0;
@@ -2859,6 +2841,7 @@ inline void load_switches(const pugi::xml_node& root, T& out, Context& context, 
         switch (in) {
             case gtok_t_switches::SWITCH: {
                 int switch_id;
+                memset(&switch_id, 0, sizeof(switch_id));
                 load_switch_required_attributes(node, &switch_id, report_error);
                 auto child_context = out.add_switches_switch(context, switch_id);
                 load_switch(node, out, child_context, report_error, offset_debug);
@@ -2895,7 +2878,7 @@ inline void load_segment_timing(const pugi::xml_node& root, T& out, Context& con
     }
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <segment_timing>.");
+        noreturn_report(report_error, "Unexpected child element in <segment_timing>.");
 }
 
 template<class T, typename Context>
@@ -2928,7 +2911,7 @@ inline void load_segment(const pugi::xml_node& root, T& out, Context& context, c
         if (gstate[(int)in] == 0)
             gstate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate element " + std::string(node.name()) + " in <segment>.").c_str());
+            noreturn_report(report_error, ("Duplicate element " + std::string(node.name()) + " in <segment>.").c_str());
         switch (in) {
             case gtok_t_segment::TIMING: {
                 auto child_context = out.init_segment_timing(context);
@@ -2959,7 +2942,7 @@ inline void load_segments(const pugi::xml_node& root, T& out, Context& context, 
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <segments>.");
+        noreturn_report(report_error, "Unexpected attribute in <segments>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t segment_count = 0;
@@ -2994,6 +2977,7 @@ inline void load_segments(const pugi::xml_node& root, T& out, Context& context, 
         switch (in) {
             case gtok_t_segments::SEGMENT: {
                 int segment_id;
+                memset(&segment_id, 0, sizeof(segment_id));
                 load_segment_required_attributes(node, &segment_id, report_error);
                 auto child_context = out.add_segments_segment(context, segment_id);
                 load_segment(node, out, child_context, report_error, offset_debug);
@@ -3017,7 +3001,7 @@ inline void load_pin(const pugi::xml_node& root, T& out, Context& context, const
 
     out.set_pin_value(root.child_value(), context);
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <pin>.");
+        noreturn_report(report_error, "Unexpected child element in <pin>.");
 }
 
 constexpr int NUM_T_PIN_CLASS_STATES = 2;
@@ -3068,6 +3052,7 @@ inline void load_pin_class(const pugi::xml_node& root, T& out, Context& context,
         switch (in) {
             case gtok_t_pin_class::PIN: {
                 int pin_ptc;
+                memset(&pin_ptc, 0, sizeof(pin_ptc));
                 load_pin_required_attributes(node, &pin_ptc, report_error);
                 auto child_context = out.add_pin_class_pin(context, pin_ptc);
                 load_pin(node, out, child_context, report_error, offset_debug);
@@ -3147,6 +3132,7 @@ inline void load_block_type(const pugi::xml_node& root, T& out, Context& context
         switch (in) {
             case gtok_t_block_type::PIN_CLASS: {
                 enum_pin_type pin_class_type;
+                memset(&pin_class_type, 0, sizeof(pin_class_type));
                 load_pin_class_required_attributes(node, &pin_class_type, report_error);
                 auto child_context = out.add_block_type_pin_class(context, pin_class_type);
                 load_pin_class(node, out, child_context, report_error, offset_debug);
@@ -3175,7 +3161,7 @@ inline void load_block_types(const pugi::xml_node& root, T& out, Context& contex
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <block_types>.");
+        noreturn_report(report_error, "Unexpected attribute in <block_types>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t block_type_count = 0;
@@ -3210,8 +3196,11 @@ inline void load_block_types(const pugi::xml_node& root, T& out, Context& contex
         switch (in) {
             case gtok_t_block_types::BLOCK_TYPE: {
                 int block_type_height;
+                memset(&block_type_height, 0, sizeof(block_type_height));
                 int block_type_id;
+                memset(&block_type_id, 0, sizeof(block_type_id));
                 int block_type_width;
+                memset(&block_type_width, 0, sizeof(block_type_width));
                 load_block_type_required_attributes(node, &block_type_height, &block_type_id, &block_type_width, report_error);
                 auto child_context = out.add_block_types_block_type(context, block_type_height, block_type_id, block_type_width);
                 load_block_type(node, out, child_context, report_error, offset_debug);
@@ -3234,7 +3223,7 @@ inline void load_grid_loc(const pugi::xml_node& root, T& out, Context& context, 
     *offset_debug = root.offset_debug();
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <grid_loc>.");
+        noreturn_report(report_error, "Unexpected child element in <grid_loc>.");
 }
 
 constexpr int NUM_T_GRID_LOCS_STATES = 2;
@@ -3253,7 +3242,7 @@ inline void load_grid_locs(const pugi::xml_node& root, T& out, Context& context,
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <grid_locs>.");
+        noreturn_report(report_error, "Unexpected attribute in <grid_locs>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t grid_loc_count = 0;
@@ -3288,10 +3277,15 @@ inline void load_grid_locs(const pugi::xml_node& root, T& out, Context& context,
         switch (in) {
             case gtok_t_grid_locs::GRID_LOC: {
                 int grid_loc_block_type_id;
+                memset(&grid_loc_block_type_id, 0, sizeof(grid_loc_block_type_id));
                 int grid_loc_height_offset;
+                memset(&grid_loc_height_offset, 0, sizeof(grid_loc_height_offset));
                 int grid_loc_width_offset;
+                memset(&grid_loc_width_offset, 0, sizeof(grid_loc_width_offset));
                 int grid_loc_x;
+                memset(&grid_loc_x, 0, sizeof(grid_loc_x));
                 int grid_loc_y;
+                memset(&grid_loc_y, 0, sizeof(grid_loc_y));
                 load_grid_loc_required_attributes(node, &grid_loc_block_type_id, &grid_loc_height_offset, &grid_loc_width_offset, &grid_loc_x, &grid_loc_y, report_error);
                 auto child_context = out.add_grid_locs_grid_loc(context, grid_loc_block_type_id, grid_loc_height_offset, grid_loc_width_offset, grid_loc_x, grid_loc_y);
                 load_grid_loc(node, out, child_context, report_error, offset_debug);
@@ -3340,7 +3334,7 @@ inline void load_node_loc(const pugi::xml_node& root, T& out, Context& context, 
     }
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <node_loc>.");
+        noreturn_report(report_error, "Unexpected child element in <node_loc>.");
 }
 
 template<class T, typename Context>
@@ -3353,7 +3347,7 @@ inline void load_node_timing(const pugi::xml_node& root, T& out, Context& contex
     *offset_debug = root.offset_debug();
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <node_timing>.");
+        noreturn_report(report_error, "Unexpected child element in <node_timing>.");
 }
 
 template<class T, typename Context>
@@ -3366,7 +3360,7 @@ inline void load_node_segment(const pugi::xml_node& root, T& out, Context& conte
     *offset_debug = root.offset_debug();
 
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <node_segment>.");
+        noreturn_report(report_error, "Unexpected child element in <node_segment>.");
 }
 
 template<class T, typename Context>
@@ -3391,7 +3385,7 @@ inline void load_meta(const pugi::xml_node& root, T& out, Context& context, cons
 
     out.set_meta_value(root.child_value(), context);
     if (root.first_child().type() == pugi::node_element)
-        (*report_error)("Unexpected child element in <meta>.");
+        noreturn_report(report_error, "Unexpected child element in <meta>.");
 }
 
 constexpr int NUM_T_METADATA_STATES = 2;
@@ -3410,7 +3404,7 @@ inline void load_metadata(const pugi::xml_node& root, T& out, Context& context, 
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <metadata>.");
+        noreturn_report(report_error, "Unexpected attribute in <metadata>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t meta_count = 0;
@@ -3491,14 +3485,19 @@ inline void load_node(const pugi::xml_node& root, T& out, Context& context, cons
         if (gstate[(int)in] == 0)
             gstate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate element " + std::string(node.name()) + " in <node>.").c_str());
+            noreturn_report(report_error, ("Duplicate element " + std::string(node.name()) + " in <node>.").c_str());
         switch (in) {
             case gtok_t_node::LOC: {
                 int node_loc_ptc;
+                memset(&node_loc_ptc, 0, sizeof(node_loc_ptc));
                 int node_loc_xhigh;
+                memset(&node_loc_xhigh, 0, sizeof(node_loc_xhigh));
                 int node_loc_xlow;
+                memset(&node_loc_xlow, 0, sizeof(node_loc_xlow));
                 int node_loc_yhigh;
+                memset(&node_loc_yhigh, 0, sizeof(node_loc_yhigh));
                 int node_loc_ylow;
+                memset(&node_loc_ylow, 0, sizeof(node_loc_ylow));
                 load_node_loc_required_attributes(node, &node_loc_ptc, &node_loc_xhigh, &node_loc_xlow, &node_loc_yhigh, &node_loc_ylow, report_error);
                 auto child_context = out.init_node_loc(context, node_loc_ptc, node_loc_xhigh, node_loc_xlow, node_loc_yhigh, node_loc_ylow);
                 load_node_loc(node, out, child_context, report_error, offset_debug);
@@ -3506,7 +3505,9 @@ inline void load_node(const pugi::xml_node& root, T& out, Context& context, cons
             } break;
             case gtok_t_node::TIMING: {
                 float node_timing_C;
+                memset(&node_timing_C, 0, sizeof(node_timing_C));
                 float node_timing_R;
+                memset(&node_timing_R, 0, sizeof(node_timing_R));
                 load_node_timing_required_attributes(node, &node_timing_C, &node_timing_R, report_error);
                 auto child_context = out.init_node_timing(context, node_timing_C, node_timing_R);
                 load_node_timing(node, out, child_context, report_error, offset_debug);
@@ -3514,6 +3515,7 @@ inline void load_node(const pugi::xml_node& root, T& out, Context& context, cons
             } break;
             case gtok_t_node::SEGMENT: {
                 int node_segment_segment_id;
+                memset(&node_segment_segment_id, 0, sizeof(node_segment_segment_id));
                 load_node_segment_required_attributes(node, &node_segment_segment_id, report_error);
                 auto child_context = out.init_node_segment(context, node_segment_segment_id);
                 load_node_segment(node, out, child_context, report_error, offset_debug);
@@ -3548,7 +3550,7 @@ inline void load_rr_nodes(const pugi::xml_node& root, T& out, Context& context, 
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <rr_nodes>.");
+        noreturn_report(report_error, "Unexpected attribute in <rr_nodes>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t node_count = 0;
@@ -3583,8 +3585,11 @@ inline void load_rr_nodes(const pugi::xml_node& root, T& out, Context& context, 
         switch (in) {
             case gtok_t_rr_nodes::NODE: {
                 unsigned int node_capacity;
+                memset(&node_capacity, 0, sizeof(node_capacity));
                 unsigned int node_id;
+                memset(&node_id, 0, sizeof(node_id));
                 enum_node_type node_type;
+                memset(&node_type, 0, sizeof(node_type));
                 load_node_required_attributes(node, &node_capacity, &node_id, &node_type, report_error);
                 auto child_context = out.add_rr_nodes_node(context, node_capacity, node_id, node_type);
                 load_node(node, out, child_context, report_error, offset_debug);
@@ -3613,7 +3618,7 @@ inline void load_edge(const pugi::xml_node& root, T& out, Context& context, cons
         if (gstate[(int)in] == 0)
             gstate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate element " + std::string(node.name()) + " in <edge>.").c_str());
+            noreturn_report(report_error, ("Duplicate element " + std::string(node.name()) + " in <edge>.").c_str());
         switch (in) {
             case gtok_t_edge::METADATA: {
                 auto child_context = out.init_edge_metadata(context);
@@ -3644,7 +3649,7 @@ inline void load_rr_edges(const pugi::xml_node& root, T& out, Context& context, 
     *offset_debug = root.offset_debug();
 
     if (root.first_attribute())
-        (*report_error)("Unexpected attribute in <rr_edges>.");
+        noreturn_report(report_error, "Unexpected attribute in <rr_edges>.");
 
     // Preallocate arrays by counting child nodes (if any)
     size_t edge_count = 0;
@@ -3679,8 +3684,11 @@ inline void load_rr_edges(const pugi::xml_node& root, T& out, Context& context, 
         switch (in) {
             case gtok_t_rr_edges::EDGE: {
                 unsigned int edge_sink_node;
+                memset(&edge_sink_node, 0, sizeof(edge_sink_node));
                 unsigned int edge_src_node;
+                memset(&edge_src_node, 0, sizeof(edge_src_node));
                 unsigned int edge_switch_id;
+                memset(&edge_switch_id, 0, sizeof(edge_switch_id));
                 load_edge_required_attributes(node, &edge_sink_node, &edge_src_node, &edge_switch_id, report_error);
                 auto child_context = out.add_rr_edges_edge(context, edge_sink_node, edge_src_node, edge_switch_id);
                 load_edge(node, out, child_context, report_error, offset_debug);
@@ -3726,7 +3734,7 @@ inline void load_rr_graph(const pugi::xml_node& root, T& out, Context& context, 
         if (gstate[(int)in] == 0)
             gstate[(int)in] = 1;
         else
-            (*report_error)(("Duplicate element " + std::string(node.name()) + " in <rr_graph>.").c_str());
+            noreturn_report(report_error, ("Duplicate element " + std::string(node.name()) + " in <rr_graph>.").c_str());
         switch (in) {
             case gtok_t_rr_graph::CHANNELS: {
                 auto child_context = out.init_rr_graph_channels(context);
@@ -4158,7 +4166,7 @@ inline void dfa_error(const char* wrong, const int* states, const char* const* l
     for (unsigned int i = 1; i < expected.size(); i++)
         expected_or += std::string(" or ") + expected[i];
 
-    (*report_error)(("Expected " + expected_or + ", found " + std::string(wrong)).c_str());
+    noreturn_report(report_error, ("Expected " + expected_or + ", found " + std::string(wrong)).c_str());
 }
 
 template<std::size_t N>
@@ -4172,7 +4180,7 @@ inline void all_error(std::bitset<N> gstate, const char* const* lookup, const st
     for (unsigned int i = 1; i < missing.size(); i++)
         missing_and += std::string(", ") + missing[i];
 
-    (*report_error)(("Didn't find required elements " + missing_and + ".").c_str());
+    noreturn_report(report_error, ("Didn't find required elements " + missing_and + ".").c_str());
 }
 
 template<std::size_t N>
@@ -4186,7 +4194,7 @@ inline void attr_error(std::bitset<N> astate, const char* const* lookup, const s
     for (unsigned int i = 1; i < missing.size(); i++)
         missing_and += std::string(", ") + missing[i];
 
-    (*report_error)(("Didn't find required attributes " + missing_and + ".").c_str());
+    noreturn_report(report_error, ("Didn't find required attributes " + missing_and + ".").c_str());
 }
 
 inline void get_line_number(const char* filename, std::ptrdiff_t target_offset, int* line, int* col) {

--- a/vpr/src/route/rr_graph_writer.cpp
+++ b/vpr/src/route/rr_graph_writer.cpp
@@ -9,6 +9,7 @@
 
 #include <cstdio>
 #include <fstream>
+#include <limits>
 #include "rr_graph_uxsdcxx_serializer.h"
 #include "rr_graph_uxsdcxx.h"
 #ifdef VTR_ENABLE_CAPNPROTO
@@ -48,9 +49,10 @@ void write_rr_graph(const char* file_name, const std::vector<t_segment_inf>& seg
         &device_ctx.rr_edge_metadata,
         &device_ctx.arch->strings);
 
-    std::fstream fp;
-    fp.open(file_name, std::fstream::out | std::fstream::trunc);
     if (vtr::check_file_name_extension(file_name, ".xml")) {
+        std::fstream fp;
+        fp.open(file_name, std::fstream::out | std::fstream::trunc);
+        fp.precision(std::numeric_limits<float>::max_digits10);
         void* context;
         uxsd::write_rr_graph_xml(reader, context, fp);
 #ifdef VTR_ENABLE_CAPNPROTO

